src/foam/dao/WriteOnlyFileJournal.js

```diff
@@ -32,8 +35,18 @@ foam.CLASS({
         final Object id = obj.getProperty("id");
 
         getLine().enqueue(new foam.util.concurrent.AbstractAssembly() {
+          FObject old;
           String record_ = null;
 
+          public Object[] requestLocks() {
+            return new Object[] { id };
```

Comment:
Why is this needed?

Reply:
requestLocks support isn't implemented yet, and if it were, this wouldn't work because you aren't guaranteed to get the exact same object every time for 'id', even if they do have the same value.